app/javascript/components/ResetPassword.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import Axios from 'axios';
+const queryString = require('query-string');
+
+export default class ResetPasswordForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const parsed = queryString.parse(this.props.location.search);
+    this.state = { user: { password: '',
+                           password_confirmation: '',
+                           'access-token': parsed['access-token'],
+                           client: parsed.client,
+                           uid: parsed.uid
+                          }
+                  };
+    this.handleResetPassword = this.handleResetPassword.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleResetPassword = (e) => {
+    e.preventDefault();
+    Axios({
+      method: 'put',
+      url: '/auth/password',
+      data: this.state.user
+    }).then(response => {
+      this.props.history.push('/sign_in');
```

Comment:
Maybe after recovery password user must be authenticated and redirect to main page